nx/lib/nx/random.ex

```diff
@@ -821,16 +827,25 @@ defmodule Nx.Random do
       while {i = 0, tensor, key}, i < num_rounds do
         keys = split(key)
         sort_keys = random_bits(keys[1], shape: tensor.shape)
-        tensor = sort_key_val(tensor, sort_keys, axis: axis)
+        tensor = sort_key_val(tensor, sort_keys, axis: axis, independent: independent)
         {i + 1, tensor, keys[0]}
       end
 
     {out, key}
   end
 
-  defnp sort_key_val(tensor, sort_keys, opts \\ []) do
+  deftransformp sort_key_val(tensor, sort_keys, opts \\ []) do
     idx = Nx.argsort(sort_keys, axis: opts[:axis])
-    Nx.take_along_axis(tensor, idx, axis: opts[:axis])
+
+    if opts[:independent] do
+      # We need to use take_along_axis in the independent case because
+      # the sort_keys tensor has the same shape as the input tensor.
+      Nx.take_along_axis(tensor, idx, axis: opts[:axis])
```

Comment:
I think `take_along_axis` with 1D tensors is supposed to work just fine, the issue must be something else.

Reply:
I wonder if this is actually a bug in EXLA's take_along_axis somehow.

The generated indices from argsort are returning with sensible values, but somehow these aren't being applied properly by take_along_axis